### PR TITLE
Bug 1225090 - [TV][2.5] Remind user to add to Apps if preview content has been opened X times

### PR DESCRIPTION
**Pull request by rickychien**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ca2848e2b8aa6c095e3a73b88e07fc2a94c0ee06

File: tv_apps/smart-system/js/app_install_manager.js [hunk 1]
```diff
@@ -338,11 +338,30 @@ var AppInstallManager = {
       this.systemBanner.show({
         id: 'preview-app-hint'
       });
+      this.previewOpened[manifestURL] = ++this.previewOpened[manifestURL] || 1;
     };
 
     var handlePreviewTerminated = () => {
       window.removeEventListener('previewterminated', handlePreviewTerminated);
-      this.uninstallPreviewApp();
+      var askAddPrompt = this.previewOpened[manifestURL] === this.openedTimes;
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: Where is the definition of `openedTimes`?

Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: Question: 
- Is it correct that we show the prompt only when the previewing counter exactly equals to `openedTimes`? -- or every time when the counter is greater than `openedTimes`? 
- Should we reset the counter?

File: tv_apps/smart-system/js/app_install_manager.js [hunk 2]
```diff
@@ -373,6 +392,24 @@ var AppInstallManager = {
...
+    return this.getPreviewAppManifestURL().indexOf(manifestURL) === -1 &&
+      installedApps.indexOf(manifestURL) !== -1;
+  },
+
+  handleAddApp: function(app) {
+    this.setPreviewAppManifestURL('');
+    this.showInstallSuccess(app);
+  },
+
+  handleAddAppCancel: function(err) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: Now I know what the purpose of `handleAddAppCancel` is.

File: tv_apps/smart-system/js/app_install_manager.js [hunk 3]
```diff
@@ -338,11 +338,30 @@ var AppInstallManager = {
...
+        var TYPES = AppInstallDialogs.TYPES;
+        // Keep app info in case of it is destroyed by terminated event
+        var cloneApp = {
+          manifestURL: app.manifestURL,
+          manifest: app.manifest,
+          updateManifest: app.updateManifest
+        };
+        this.appInstallDialogs.show(TYPES.AddAppDialog, options).then(
+          this.handleAddApp.bind(this, cloneApp),
+          this.handleAddAppCancel.bind(this)
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: I guess we don't really need the `err` message when cancelling. How about passing `this.uninstallPreviewApp.bind(this)` directly here?

Comment (rickychien) [inline tv_apps/smart-system/js/app_install_manager.js]: I will use an empty callback

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=57819cf003a09561d63f7aa7089c1f506772265b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0af42e978455afb6878769f400c271baf7f9b026

File: tv_apps/smart-system/js/app_install_manager.js [hunk 4]
```diff
@@ -17,7 +17,10 @@
 // note: we removed the download cancel dialog because we don't have an UI to
 // remove it.
 
+const PREVIEW_OPENED_TIMES = 3;
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: nit `PREVIEW_OPEN_TIMES_TO_HINT`

File: tv_apps/smart-system/js/app_install_manager.js [hunk 5]
```diff
@@ -17,7 +17,10 @@
...
+
 var AppInstallManager = {
+
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: no need this blank line to reduce modification.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5473cdd12c5970519931e2408929baeb1df1acb2

File: tv_apps/smart-system/js/app_install_manager.js [hunk 6]
```diff
@@ -45,6 +48,7 @@ var AppInstallManager = {
     this.appInfos = {};
     this.setupQueue = [];
     this.isSetupInProgress = false;
+    this.previewOpened = {};
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_install_manager.js]: nit: `previewOpenedTimes`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c5f4a5144328d2866cbf61d343ae18920c9731c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4100dbc261814c89693ceee5208e48da18caa121

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=675feebd2a0a194620198cedb44fbd9f4bf3bfcd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c9c11b07d27fe4186a59902093c2bc8ae6948732

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c6bf10bcf30c30724c27b0fe02e8bc0b6016474a

**Pull request merged**